etc/kayobe/kolla/globals.yml

```
@@ -7,11 +7,19 @@ kolla_base_distro: "{% raw %}{{ ansible_facts.distribution | lower }}{% endraw %
 
 # Use facts so this is determined correctly when the control host OS differs
 # from os_distribuition.
-kolla_base_distro_version: "{% raw %}{{ kolla_base_distro_version_default_map[kolla_base_distro] }}{% endraw %}"
+kolla_base_distro_version: "{% raw %}{{ 'noble' if ansible_facts.distribution_release == 'noble' else kolla_base_distro_version_default_map[kolla_base_distro] }}{% endraw %}"
 
 # Convenience variable for base distro and version string.
 kolla_base_distro_and_version: "{% raw %}{{ kolla_base_distro }}-{{ kolla_base_distro_version }}{% endraw %}"
 
+kolla_base_distro_version_custom_map: {
+  "rocky-9": "3.9",
+  "ubuntu-jammy": "3.10",
+  "ubuntu-noble": "3.12"
+}
```

I can see why the python versions might need to be quoted to make sure they are strings not float, but does the rest need to be quoted and bracketed?
```suggestion
kolla_base_distro_version_custom_map:
  rocky-9: "3.9"
  ubuntu-jammy: "3.10"
  ubuntu-noble: "3.12"
```

---

I followed the style of upstream Kolla-ansible: https://opendev.org/openstack/kolla-ansible/src/branch/stable/2024.1/ansible/group_vars/all.yml#L1255-L1260